genie-web/src/main/java/com/netflix/genie/web/services/JobOwnerService.java

```diff
@@ -0,0 +1,44 @@
+/*
+ *
+ *  Copyright 2016 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.genie.web.services;
+
+import com.netflix.genie.common.exceptions.GenieException;
+import org.springframework.validation.annotation.Validated;
+
+import javax.validation.constraints.NotBlank;
+
+/**
+ * Interface for getting information about the host owning the job.
+ *
+ * @author standon
+ * @since 4.0.0
+ */
+@Validated
+public interface JobOwnerService {
+
+    /**
+     * Get the hostName owning the job.
+     *
+     * @param jobId job id
+     * @return hostname of the job owner
+     * @throws GenieException if there is an error finding the host owning the job
+     */
+    String getJobOwnerHostName(
+        @NotBlank(message = "No job id entered. Unable to find the owner.") final String jobId
+    ) throws GenieException;
```

Comment:
This is that slippery slope to generic exception throwing.

Reply:
I would like to throw `GenieNotFoundException` here. Here is my problem. In this method I am using `jobSearchService.getJobHost`. It throws a `GenieException`. Internally its only throwing a `GenieNotFoundException`. So, either we change the `jobSearchService.getJobHost` to throw `GenieNotFoundException` or I catch `GenieException` in this method and convert it to throw `GenieNotFoundException`. Thoughts ?